crates/aptos/src/account/key_rotation.rs

```diff
@@ -38,31 +36,57 @@ use std::{collections::BTreeMap, path::PathBuf};
 /// rotated you will need to use the original account address, with the
 /// new private key.  There is an interactive prompt to help you add it
 /// to a new profile.
+///
+/// If you wish to rotate from a ledger wallet, it must have its own
+/// profile. If you wish to rotate to a ledger wallet, specify the new
+/// derivation path or index accordingly.
 #[derive(Debug, Parser)]
 pub struct RotateKey {
     #[clap(flatten)]
     pub(crate) txn_options: TransactionOptions,
 
+    #[clap(flatten)]
+    pub(crate) new_auth_key_options: NewAuthKeyOptions,
+
+    #[clap(flatten)]
+    pub(crate) new_profile_options: NewProfileOptions,
+}
+
+#[derive(Args, Debug)]
+#[group(required = true, multiple = false)]
+pub(crate) struct NewAuthKeyOptions {
     /// File name that contains the new private key encoded in the type from `--encoding`
-    #[clap(long, group = "new_private_key_inputs", value_parser)]
+    #[clap(long, value_parser)]
     pub(crate) new_private_key_file: Option<PathBuf>,
 
     /// New private key encoded in the type from `--encoding`
-    #[clap(long, group = "new_private_key_inputs")]
+    #[clap(long)]
     pub(crate) new_private_key: Option<String>,
 
-    /// Name of the profile to save the new private key
+    /// BIP44 derivation path of hardware wallet account, e.g. `m/44'/637'/0'/0'/0'`
     ///
-    /// If not provided, it will interactively have you save a profile,
-    /// unless `--skip_saving_profile` is provided
+    /// Note you may need to escape single quotes in your shell, for example
+    /// `m/44'/637'/0'/0'/0'` would be `m/44\'/637\'/0\'/0\'/0\'`
     #[clap(long)]
-    pub(crate) save_to_profile: Option<String>,
+    pub(crate) new_derivation_path: Option<String>,
 
-    /// Skip saving profile
+    /// BIP44 account index of hardware wallet account, e.g. `0`
     ///
-    /// This skips the interactive profile saving after rotating the authentication key
+    /// Given index `n` maps to BIP44 derivation path `m/44'/637'/n'/0'/0`
+    #[clap(long)]
+    pub(crate) new_derivation_index: Option<String>,
+}
+
+#[derive(Args, Debug)]
+#[group(required = true, multiple = false)]
+pub(crate) struct NewProfileOptions {
+    /// Only specify if you do not want to save a new profile
     #[clap(long)]
     pub(crate) skip_saving_profile: bool,
```

Comment:
I know this feature already existed but when would you want to do this? cc @gregnazario

Reply:
> I know this feature already existed 

Right, I don't want to break compatibility so I'm just trying to make sure the grouping semantics are workable (though I still think it could be useful, see elow)

> when would you want to do this

1. You just want an example of a transaction
1. You are running an e2e test on a persisted runner and don't want to clutter the config
1. You are rotating back to a ledger after temporarily rotating to a hot wallet in order to publish a transaction, and you don't want to clutter up your config because elsewhere you have a database of which derivation path should map to which wallet